app/models/task.py

```diff
@@ -1,5 +1,73 @@
 from app import db
-
+from datetime import datetime
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    '''
+        A task to be done.
+
+        Expected elements on creation:
+            title (str): name of task
+            description (str): task description
+        Optional elements:
+            completed_at (datetime): timestamp of task completion
+            goal_id (int): id of associated goal
+    '''
+    ## class variables
+    expected_elements = {"title":str, "description":str}
+    db_name = "Task"
+
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, default=None)
+    goal_id = db.Column(db.Integer, db.ForeignKey("goal.goal_id")) #note: need to make nullable? or that way by default?
+    goal = db.relationship("Goal", back_populates="tasks") #lazy=select/True is default
+
+    def make_response_dict(self):
+        '''Build and return dict of Task instance attributes'''
+        if self.completed_at:
+            completed = True
+        else:
+            completed = False
```

Comment:
This logic can be simplified using the `bool` function:

```suggestion
        completed = bool(self.completed_at)
```